C++ plugin template on Windows broken by CMake definitions

**Version and Platform (required):**
- Binary Ninja Version: Since https://github.com/Vector35/binaryninja-api/commit/0d031494987b6e1f36ebaaf3b0b92c278cd8ca6b
- Edition: Commercial
- OS: Windows
- OS Version: 11
- CPU Architecture: x64

**Bug Description:**

Copying text from commit comment https://github.com/Vector35/binaryninja-api/commit/0d031494987b6e1f36ebaaf3b0b92c278cd8ca6b#commitcomment-163348146

> Why is this change needed? Shouldn't the compiler and/or CMake handle setting these automatically?
> 
> `_DEBUG` isn't a standardized cross-platform macro, whereas `NDEBUG` is implemented by all compilers and checked in `assert.h` https://stackoverflow.com/questions/2290509/debug-vs-ndebug
> 
> This causes Windows failures while trying to build "RelWithDebInfo" using the template repo https://github.com/Vector35/sample_plugin_cpp
> 
> https://github.com/ekilmer/binja-test-cpp-plugin/actions/runs/16708139292/job/47289131882#step:8:171
> 
> ```
> binaryninjaapi.lib(fileaccessor.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MD_DynamicRelease' in library.cpp.obj
> binaryninjaapi.lib(linearviewobject.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '2' doesn't match value '0' in library.cpp.obj
> binaryninjaapi.lib(linearviewobject.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MD_DynamicRelease' in library.cpp.obj
> binaryninjaapi.lib(workflow.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '2' doesn't match value '0' in library.cpp.obj
> binaryninjaapi.lib(workflow.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MD_DynamicRelease' in library.cpp.obj
> ```

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Use the template plugin repo to build for Windows (latest commit https://github.com/Vector35/sample_plugin_cpp/tree/1452996918516482116119499764549231183048)

**Expected Behavior:**
Windows builds without failure _and_ I can still select any of the four default CMake build options: `Release`, `Debug`, `RelWithDebInfo`, `MinSizeRel`.

**Screenshots/Video Recording:**

See bug description

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,9 +63,7 @@ if(BN_REF_COUNT_DEBUG)
 endif()
 
 if(CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES RelWithDebInfo)
-    target_compile_definitions(binaryninjaapi PRIVATE _DEBUG)
-else()
-    target_compile_definitions(binaryninjaapi PRIVATE NDEBUG)
+    target_compile_definitions(binaryninjaapi PRIVATE BN_ENABLE_LOG_TRACE)
 endif()
 
 add_subdirectory(vendor/fmt EXCLUDE_FROM_ALL)

--- a/log.cpp
+++ b/log.cpp
@@ -215,7 +215,7 @@ void BinaryNinja::Log(BNLogLevel level, const char* fmt, ...)
 
 void BinaryNinja::LogTrace(const char* fmt, ...)
 {
-#ifdef _DEBUG
+#ifdef BN_ENABLE_LOG_TRACE
 	va_list args;
 	va_start(args, fmt);
 	PerformLog(0, DebugLog, "", 0, fmt, args);
@@ -280,7 +280,7 @@ void BinaryNinja::LogForException(BNLogLevel level, const std::exception& e, con
 
 void BinaryNinja::LogTraceForException(const std::exception& e, const char* fmt, ...)
 {
-#ifdef _DEBUG
+#ifdef BN_ENABLE_LOG_TRACE
 	va_list args;
 	va_start(args, fmt);
 	PerformLogForException(0, DebugLog, "", 0, e, fmt, args);
@@ -345,12 +345,12 @@ void BinaryNinja::LogWithStackTrace(BNLogLevel level, const char* fmt, ...)
 
 void BinaryNinja::LogTraceWithStackTrace(const char* fmt, ...)
 {
-	#ifdef _DEBUG
+#ifdef BN_ENABLE_LOG_TRACE
 	va_list args;
 	va_start(args, fmt);
 	PerformLogWithStackTrace(0, DebugLog, "", 0, fmt, args);
 	va_end(args);
-	#endif
+#endif
 }
 
 
@@ -493,7 +493,7 @@ void Logger::Log(BNLogLevel level, const char* fmt, ...)
 
 void Logger::LogTrace(const char* fmt, ...)
 {
-#ifdef _DEBUG
+#ifdef BN_ENABLE_LOG_TRACE
 	va_list args;
 	va_start(args, fmt);
 	PerformLog(GetSessionId(), DebugLog, GetName(), GetThreadId(), fmt::format("{}{}", GetIndent(), fmt).c_str(), args);
@@ -559,7 +559,7 @@ void Logger::LogForException(BNLogLevel level, const std::exception& e, const ch
 
 void Logger::LogTraceForException(const std::exception& e, const char* fmt, ...)
 {
-#ifdef _DEBUG
+#ifdef BN_ENABLE_LOG_TRACE
 	va_list args;
 	va_start(args, fmt);
 	PerformLogForException(
@@ -631,13 +631,13 @@ void Logger::LogWithStackTrace(BNLogLevel level, const char* fmt, ...)
 
 void Logger::LogTraceWithStackTrace(const char* fmt, ...)
 {
-	#ifdef _DEBUG
+#ifdef BN_ENABLE_LOG_TRACE
 	va_list args;
 	va_start(args, fmt);
 	PerformLogWithStackTrace(
 		GetSessionId(), DebugLog, GetName(), GetThreadId(), fmt::format("{}{}", GetIndent(), fmt).c_str(), args);
 	va_end(args);
-	#endif
+#endif
 }
 
 

